### PR TITLE
Track preference options usage

### LINKED ISSUE #6791: track preference options usage
/kind user-story

## User Story
As an odo developer, I want to know what preferences users set and to what values

## Acceptance Criteria 
- [ ] when users execute `odo preference set <parameter> <value>` command with telemetry enabled odo should send an event with `{parameter: <parameter>, value: <value>`} properties
- [ ] when users execute `odo preference unset <parameter>` command with telemetry enabled odo should send an event with `{parameter: <parameter>} property
- [ ] by default, values are anonymized, and hard-coded parameters are sent with clear value (for now, all parameters can be added to the list of non sensitive parameters)
- [ ] anonymize values with algo which permits to check for repetitive values

/kind user-story
/priority high

### PR DESCRIPTION
**Pull request by rm3l**
**What type of PR is this:**
/kind feature
/area telemetry

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**
Fixes #6791 

**PR acceptance criteria:**

- [ ] Unit test 

- [x] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: pkg/segment/context/context.go [hunk 1]
```diff
@@ -41,6 +45,10 @@ const (
 	JBoss    = "jboss"
 )
 
+// By default, all values are anomized.
+// Add the (case-insensitive) preference parameter name here to have the corresponding value sent verbatim to telemetry.
+var clearTextPreferenceParams = []string{}
```
Comment (feloy) [inline pkg/segment/context/context.go:50]: Don't we want to send boolean parameters (UpdateNotification, ConsentTelemetry, Ephemeral) in clear form? Hash value will be very easily reversed, we might as well have the clear value.

For numeric values (*Timeout, RegistryCacheTime), I think we could also send numeric value, as there can be nothing nothing personal IMO.

**Review by feloy — APPROVED** [review]

**Comment by rm3l** [comment]
```
  [FAILED] Expected
      <*url.Error | 0xc000533410>: {
          Op: "Post",
          URL: "http://127.0.0.1:34017/api/newuser",
          Err: <*errors.errorString | 0xc000072130>{s: "EOF"},
      }
  to be nil
  In [It] at: /go/odo_1/tests/e2escenarios/e2e_test.go:431 @ 05/26/23 14:48:08.161
...
------------------------------

Summarizing 1 Failure:
  [FAIL] E2E Test starting with non-empty Directory add Binding [It] should verify developer workflow of using binding as env in innerloop
  /go/odo_1/tests/e2escenarios/e2e_test.go:431

Ran 4 of 4 Specs in 257.629 seconds
FAIL! -- 3 Passed | 1 Failed | 0 Pending | 0 Skipped
```

Flaky E2E test - tracked in #6582 

/override OpenShift-Integration-tests/OpenShift-Integration-tests

**Comment by rm3l** [comment]
Force-pushed to fix Git conflicts with the `main` branch.

**Review by feloy — APPROVED** [review]

**Comment by feloy** [comment]
/override OpenShift-Integration-tests/OpenShift-Integration-tests

```
[FAILED] [82.111 seconds]
E2E Test starting with non-empty Directory add Binding [It] should verify developer workflow of using binding as env in innerloop
/go/odo_1/tests/e2escenarios/e2e_test.go:462
```

**Pull request merged**